openslides/assignments/views.py

```diff
@@ -242,6 +243,134 @@ def get_serializer_class(self):
+            self.permission_denied(request)
+        if not assignment.is_candidate(request.user):
+            raise ValidationError({'detail': _('You are not a candidate of this election.')})
+        assignment.set_blocked(request.user)
+        return _(
+            'You have withdrawn your candidature successfully. '
+            'You can not be nominated by other participants anymore.')
+
+    def get_other_user(self, request):
+        """
+        Helper method to get a specific user from request data (not the
+        request.user) so that the views self.candidature_other or
+        self.mark_elected can play with him.
+        """
+        if not isinstance(request.data, dict):
+            raise ValidationError({'detail': _('Invalid data. Expected dictionary, got %s.') % type(request.data)})
+        user_str = request.data.get('user', '')
+        try:
+            user_pk = int(user_str)
+        except ValueError:
+            raise ValidationError({'detail': _('Invalid data. Expected something like {"user": <id>}.')})
+        try:
+            user = User.objects.get(pk=user_pk)
+        except User.DoesNotExist:
+            raise ValidationError({'detail': _('Invalid data. User %d does not exist.') % user_pk})
+        return user
+
+    @detail_route(methods=['post', 'delete'])
+    def candidature_other(self, request, pk=None):
+        """
+        View to nominate other users (POST) or delete their candidature
+        status (DELETE). The client has to send {'user': <id>}.
+        """
+        if not request.user.has_perm('assignments.can_nominate_other'):
+            self.permission_denied(request)
+        user = self.get_other_user(request)
+        assignment = self.get_object()
+        if assignment.is_elected(user):
+            raise ValidationError({'detail': _('User %s is already elected.') % user})
+        if request.method == 'POST':
+            message = self.nominate_other(request, user, assignment)
+        else:
+            # request.method == 'DELETE'
+            message = self.delete_other(request, user, assignment)
+        return Response({'detail': message})
+
+    def nominate_other(self, request, user, assignment):
+        if assignment.phase == assignment.PHASE_FINISHED:
+            raise ValidationError({'detail': _('You can not nominate someone to this election because it is finished.')})
+        if assignment.phase == assignment.PHASE_VOTING and not request.user.has_perm('assignments.can_manage'):
+            # To nominate other during voting you have to be a manager.
+            self.permission_denied(request)
+        if assignment.is_candidate(user):
+            raise ValidationError({'detail': _('User %s is already a candidate of this election.') % user})
+        if assignment.is_blocked(user) and not request.user.has_perm('assignments.can_manage'):
+            raise ValidationError({'detail': _('User %s does not want to be an candidate.') % user})
```

Comment:
It seems, that an elected user can be unelected by any user by calling this method